src/tls_stream.rs

```diff
@@ -617,16 +617,12 @@ impl<S> TlsStream<S>
     }
 
     fn write_out(&mut self) -> io::Result<usize> {
-        let mut out = 0;
-        while self.out_buf.position() as usize != self.out_buf.get_ref().len() {
-            let position = self.out_buf.position() as usize;
-            let nwritten = try!(self.stream.write(&self.out_buf.get_ref()[position..]));
-            out += nwritten;
-            self.out_buf.set_position((position + nwritten) as u64);
-        }
-
-        Ok(out)
-    }
+        let mut nwritten = 0;
+		let position = self.out_buf.position() as usize;
+		nwritten += self.stream.write(&self.out_buf.get_ref()[position..])?;
```

Comment:
> A call to write represents at most one attempt to write to any wrapped object.

This isn't well formulated in the current docs, but the intention of it is
of `Write` to be as low level as the implementation allows.
Which is exactly what it is in that case (so it's fine to do repeated writes there).

https://internals.rust-lang.org/t/std-io-write-a-call-to-write-represents-at-most-one-attempt-to-write-to-any-wrapped-object/5068